### ISSUE TITLE
I want to compare sharpe ratio between my random weights and optimal weights

### ISSUE THREAD

**Issue by Younghoon-Lee** [robertmartin8/PyPortfolioOpt#358]
**What are you trying to do?**

Like I said in the title, I want to compare sharpe ratio between my random weights and optimal weights.

**What have you tried?**

Here is my code for better explanation.
(df is my 1-year stock price data frame )

```
mu = expected_returns.capm_return(df)
S = risk_models.CovarianceShrinkage(df).ledoit_wolf()
ef = EfficientFrontier(mu, S)
weights = ef.max_sharpe()
```
Then I changed _weights_ to numpy array.
Also I made random weights(_my_weights_), which is also numpy array. 
```
optimalPortfolioReutrnsMean = np.dot(np.array(mu), weights)
myPortfolioReturnsMean = np.dot(np.array(mu), my_weights)

optimalPortfolioVolatility = np.sqrt(np.dot(weights.T, np.dot(S, weights)))
myPortfolioVolatility = np.sqrt(np.dot(my_weights.T, np.dot(S, my_weights)))

optimalPortfolioSharpeRatio = optimalPortfolioReturnsMean/optimalPortfolioVolatility
myPortfolioSharpeRatio = myPortfolioReturnsMean/myPortfolioVolatility
```
I assumed that risk-free rate is zero.

Theoretically, _optimalPortfolioSharpeRatio_ has to outscore _myPortfolioSharpeRatio_ (at least same).
But I found some cases that _myPortfolioSharpeRatio_ outscores _optimalPortfolioSharpeRatio_

Is there anything I am missing out?

**Comment by phschiele** [comment]
@Younghoon-Lee You are right, in theory, the weights returned by `max_sharpe()` should outperform any other weight vector within the same bounds. Note that the function uses `risk_free_rate=0.02` as a default. Can you pass `risk_free_rate=0.0` as an argument and repeat your analysis?
